Automattic/vip-go-mu-plugins
# Search: Cache existence of Custom Search Results and return early in filter__ep_enable_do_weighting for existence of CSR

## Description
This PR does two things:
- Caches the existence of Custom Search Results (CSR) in an option
- Bails out of filter__ep_enable_do_weighting under two conditions:
1) if CSR is active and exists since CSR won't work as expected with weighting disabled
2) if CSR is active and the pointer preview display is shown for accuracy since weighting will be enabled

## Changelog Description

### Plugin Updated: Enterprise Search

Cache existence of Custom Search Results and do not use filter__ep_enable_do_weighting if Custom Search Results are used

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
1) On a fresh install with no weighting, add a custom search result.
2) Search for the custom search result and notice it doesn't return as expected.
3) Apply PR and repeat step 2.
4) Check if the option is set `vip_custom_results_existence` to `"1"`
